src/dio/detect_format.cpp

```diff
@@ -21,6 +21,7 @@
 #define GIF_89_STAMP     "GIF89a"
 #define PNG_MAGIC_DWORD1 0x474E5089
 #define PNG_MAGIC_DWORD2 0x0A1A0A0D
+//#define TIFF_MAGIC_DWORD 0x     //TODO implement byte check
```

Comment:
[According to Wikipedia](https://en.wikipedia.org/wiki/List_of_file_signatures) there are two possible magic dwords (the first for little-endian images, the second for big-endian images):
```suggestion
#define TIFF_MAGIC_DWORD1 0x49492A00
#define TIFF_MAGIC_DWORD2 0x4D4D002A
```
And then you'll need something like this to actually detect the format based on those:
```diff
--- src/dio/detect_format.cpp.orig	2022-04-01 11:56:58.000000000 -0500
+++ src/dio/detect_format.cpp	2022-04-01 11:58:51.000000000 -0500
@@ -48,29 +48,35 @@
    (buf[offset+3] == ((dword & 0xff000000) >> 24)))
 
   if (n >= 2) {
-    if (n >= 6) {
-      if (n >= 8) {
-        if (n >= 12) {
-          if (std::strncmp((const char*)buf, WEBP_STAMP_1, 4) == 0 ||
-              std::strncmp((const char*)buf+8, WEBP_STAMP_2, 4) == 0)
-            return FileFormat::WEBP_ANIMATION;
+    if (n >= 4) {
+      if (n >= 6) {
+        if (n >= 8) {
+          if (n >= 12) {
+            if (std::strncmp((const char*)buf, WEBP_STAMP_1, 4) == 0 ||
+                std::strncmp((const char*)buf+8, WEBP_STAMP_2, 4) == 0)
+              return FileFormat::WEBP_ANIMATION;
+          }
+
+          if (IS_MAGIC_DWORD(0, PNG_MAGIC_DWORD1) &&
+              IS_MAGIC_DWORD(4, PNG_MAGIC_DWORD2))
+            return FileFormat::PNG_IMAGE;
         }
 
-        if (IS_MAGIC_DWORD(0, PNG_MAGIC_DWORD1) &&
-            IS_MAGIC_DWORD(4, PNG_MAGIC_DWORD2))
-          return FileFormat::PNG_IMAGE;
+        if (std::strncmp((const char*)buf, GIF_87_STAMP, 6) == 0 ||
+            std::strncmp((const char*)buf, GIF_89_STAMP, 6) == 0)
+          return FileFormat::GIF_ANIMATION;
+
+        if (IS_MAGIC_WORD(4, ASE_MAGIC_NUMBER))
+          return FileFormat::ASE_ANIMATION;
+
+        if (IS_MAGIC_WORD(4, FLI_MAGIC_NUMBER) ||
+            IS_MAGIC_WORD(4, FLC_MAGIC_NUMBER))
+          return FileFormat::FLIC_ANIMATION;
       }
 
-      if (std::strncmp((const char*)buf, GIF_87_STAMP, 6) == 0 ||
-          std::strncmp((const char*)buf, GIF_89_STAMP, 6) == 0)
-        return FileFormat::GIF_ANIMATION;
-
-      if (IS_MAGIC_WORD(4, ASE_MAGIC_NUMBER))
-        return FileFormat::ASE_ANIMATION;
-
-      if (IS_MAGIC_WORD(4, FLI_MAGIC_NUMBER) ||
-          IS_MAGIC_WORD(4, FLC_MAGIC_NUMBER))
-        return FileFormat::FLIC_ANIMATION;
+      if (IS_MAGIC_DWORD(0, TIFF_MAGIC_DWORD1) ||
+          IS_MAGIC_DWORD(0, TIFF_MAGIC_DWORD2))
+        return FileFormat::TIFF_IMAGE;
     }
 
     if (IS_MAGIC_WORD(0, BMP_MAGIC_NUMBER))
```